### PR TITLE
SCALRCORE-9388 Cost Manager > Cloud Inventory > VMware > Add location filter for images, volumes, snapshots, nodes

### PR DESCRIPTION
**Pull request by mermoldy**
### Description

Replace this with the PR description (mention the changes you have made, why
you have made them, provide some background and any references to the provider
documentation if needed, etc.).

For more information on contributing, please see [Contributing](http://libcloud.readthedocs.org/en/latest/development.html#contributing)
section of our documentation.

### Status

Replace this: describe the PR status. Examples:

- work in progress
- done, ready for review

### Checklist (tick everything that applies)

- [ ] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [ ] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

File: libcloud/compute/drivers/vmware.py [hunk 1]
```diff
@@ -697,16 +723,54 @@ def _walk_snapshot_tree(self, snapshot_tree):
...
+        if url and isinstance(url, list):
+            url = url[0]
+        if isinstance(url, vim.vm.ConfigInfo.DatastoreUrlPair):
+            url = str(url.url)
+        if not url:
+            return
+
+        try:
+            url = url.split('/')
+            volume_id = url[url.index('volumes') + 1]
```
Comment (uty) [inline libcloud/compute/drivers/vmware.py:771]: Нужен комментом пример урла.

File: libcloud/compute/drivers/vmware.py [hunk 2]
```diff
@@ -1067,12 +1123,14 @@ def _to_node(self, vm_entity, vm_properties=None, datacenter_stores=None):
             extra=extra)
         return node
 
-    def _to_image(self, vm_entity, vm_properties=None):
+    def _to_image(self, vm_entity, vm_properties=None, datacenter_stores=None):
```
Comment (uty) [inline libcloud/compute/drivers/vmware.py]: Параметр `datacenter_stores`, видимо, не нужен.

**Review by uty — CHANGES_REQUESTED** [review]

File: libcloud/compute/drivers/vmware.py (hunk 2 above)
Comment (mermoldy) [reply libcloud/compute/drivers/vmware.py]: @uty убрал

File: libcloud/compute/drivers/vmware.py (hunk 1 above)
Comment (mermoldy) [reply libcloud/compute/drivers/vmware.py:771]: @uty Добавил примеры в докстринг

**Review by uty — APPROVED** [review]

**Pull request merged**